src/main/java/mate/academy/config/SecurityConfig.java

```diff
@@ -1,25 +1,55 @@
 package mate.academy.config;
 
+import mate.academy.model.Role;
+import mate.academy.security.jwt.JwtConfigurer;
+import mate.academy.security.jwt.JwtTokenProvider;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private static String ROLE_ADMIN = Role.RoleName.ADMIN.name();
+    private static String ROLE_USER = Role.RoleName.USER.name();
     private final UserDetailsService userDetailsService;
     private final PasswordEncoder passwordEncoder;
+    private final JwtTokenProvider jwtTokenProvider;
 
     public SecurityConfig(UserDetailsService userDetailsService,
-                          PasswordEncoder passwordEncoder) {
+                          PasswordEncoder passwordEncoder, JwtTokenProvider jwtTokenProvider) {
         this.userDetailsService = userDetailsService;
         this.passwordEncoder = passwordEncoder;
+        this.jwtTokenProvider = jwtTokenProvider;
     }
 
     @Override
     protected void configure(AuthenticationManagerBuilder auth) throws Exception {
         auth.userDetailsService(userDetailsService)
                 .passwordEncoder(passwordEncoder);
     }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .httpBasic().disable()
+                .csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .authorizeRequests()
+                .antMatchers("/registration", "/login", "/inject", "/").permitAll()
+                .antMatchers(HttpMethod.GET, "/products", "/users/*").hasRole(ROLE_USER)
+                .antMatchers(HttpMethod.POST, "/products").hasRole(ROLE_USER)
+                .antMatchers(HttpMethod.DELETE).hasRole(ROLE_ADMIN)
```

Comment:
```suggestion
                .antMatchers("/registration", "/login", "/inject", "/").permitAll()
                .antMatchers(HttpMethod.GET, "/products").hasRole("USER")
                .antMatchers("/products/**").hasRole("ADMIN")
                .antMatchers("/users/**").hasRole("ADMIN")
```
it is bad practice to give access to the controller  "/users/**" to users with the USER role

Reply:
Thank you
I will take into account